extensions/wikia/PortableInfobox/tests/PortableInfoboxMobileRenderServiceTest.php

```diff
@@ -0,0 +1,621 @@
+				'output' => '',
+				'description' => 'Empty data should yield no infobox markup'
+			],
+			[
+				'input' => [
+					[
+						'type' => 'title',
+						'data' => [
+							'value' => 'Test Title'
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<h2 class="pi-item pi-item-spacing pi-title">Test Title</h2>
+							</aside>',
+				'description' => 'Only title'
+			],
+			[
+				'input' => [
+					[
+						'type' => 'navigation',
+						'data' => [
+							'value' => 'navigation value',
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<nav class="pi-navigation pi-item-spacing pi-secondary-background pi-secondary-font">navigation value</nav>
+							</aside>',
+				'description' => 'navigation only'
+			],
+			[
+				'input' => [
+					[
+						'type' => 'data',
+						'data' => [
+							'label' => 'test label',
+							'value' => 'test value'
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<div class="pi-item pi-data pi-item-spacing pi-border-color">
+									<h3 class="pi-data-label pi-secondary-font">test label</h3>
+									<div class="pi-data-value pi-font">test value</div>
+								</div>
+							</aside>',
+				'description' => 'Only pair'
+			],
+			[
+				'input' => [
+					[
+						'type' => 'title',
+						'data' => [
+							'value' => 'Test Title'
+						]
+					],
+					[
+						'type' => 'image',
+						'data' => [
+							[
+								'alt' => 'image alt',
+								'url' => 'http://image.jpg',
+								'name' => 'image',
+								'key' => 'image',
+								'isVideo' => false
+							]
+						]
+					],
+					[
+						'type' => 'data',
+						'data' => [
+							'label' => 'test label',
+							'value' => 'test value'
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<h2 class="pi-item pi-item-spacing pi-title">Test Title</h2>
+								<div class="pi-item pi-image">
+    								<img src="data:image/gif;base64,R0lGODlhAQABAIABAAAAAP///yH5BAEAAAEALAAAAAABAAEAQAICTAEAOw%3D%3D" data-src="http://image.jpg" class="pi-image-thumbnail lazy media article-media" alt="image alt" data-image-key="test1" data-image-name="test1" data-ref="1" data-params="[{&quot;name&quot;:&quot;test1&quot;, &quot;full&quot;:&quot;http://image.jpg&quot;}]"/>
+								</div>
+								<div class="pi-item pi-data pi-item-spacing pi-border-color">
+									<h3 class="pi-data-label pi-secondary-font">test label</h3>
+									<div class="pi-data-value pi-font">test value</div>
+									</div>
+							</aside>',
+				'description' => 'Simple infobox with title, image and key-value pair',
+				'mockParams' => [
+					'extendImageData' => [
+						'alt' => 'image alt',
+						'url' => 'http://image.jpg',
+						'name' => 'test1',
+						'key' => 'test1',
+						'ref' => 1,
+						'width' => '400',
+						'height' => '200',
+						'thumbnail' => 'http://image.jpg',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "thumbnail" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)